Functions/Common/Get-ItemType.ps1

```diff
@@ -0,0 +1,13 @@
+﻿function Get-ItemType
+{
+        param(
+            [string]$FileExtension
+        )
+switch ($FileExtension)
+    {
+        '.rdl'  {return 'Report'}
+        '.rsds' {return 'DataSource'}
+        '.rsd'  {return 'DataSet'}
+        default {throw 'Uploading currently only supports .rdl, .rsds and .rsd files'}
```

Comment:
Now that is a generic function I would change the exception text to not mention upload